### DISCUSSION TITLE
Go forward and Go back icons?

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#785] [Ideas]
In earlier iterations of FP, I could swear that the Go Forward and Go Back functions were accessible through the toolbar. That no longer seems to be the case. Is there any way that they could be returned to the toolbar? If my memory is flawed, consider this to be a feature request.

~ Bal

**Comment by macmarrum** [comment]
Recently, there was a similar request for a toolbar button/icon to create a new map
- #747 

The toolbars and menu in Freeplane can be customized. You can add the buttons to go forward and back.

Copy the file `<install-dir>/resources/xml/mindmapmodemenu.xml` to `<user-dir>/resources/xml/mindmapmodemenu.xml`. You can use `Tools->Open user directory`. You might need to create `resources` and `xml` directories.

In the copied file, in the "main_toolbar", you can add an entry with `ForwardAction` and another entry with `BackAction` or replace some existing actions.

Here's an excerpt from my `mindmapmodemenu.xml`, after I added the back and forward buttons:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<FreeplaneUIEntries>
  <Entry name="main_toolbar" builder="toolbar,ignore">
    <Entry builder="separator" />
    <Entry row="1">
      <Entry name="OpenAction" />
    </Entry>
    <Entry row="2">
      <Entry name="SaveAction" />
    </Entry>
    <Entry row="1">
      <Entry name="BackAction" />
      <Entry name="ForwardAction" />
    </Entry>
    <Entry row="2" usedBy="EDITOR">
      <Entry name="UndoAction" />
      <Entry name="RedoAction" />
    </Entry>
    <Entry builder="separator" />
```

**Reply by wordmuse** [reply]
I'm just seeing this now... When I download the installation file, it's an _.exe_ file. Where would I find the <install-dir>. Should I download the new version as a zip file?  Also, whenever there is an update, would I have to go through this process each time, or will my choice stick?

Thanks much.

~ Bal

**Reply by macmarrum** [reply]
> Should I download the new version as a zip file?

There is no need to download anything. All is already available in the directory where Freeplane was installed. On MS Windows it's most probably `C:\Program Files\Freeplane`.

Copy the file `C:\Program Files\Freeplane\resources\xml\mindmapmodemenu.xml` and paste it in `<user-dir>\resources\xml\`. You can use Tools->Open user directory. You might need to create resources and xml directories.

> whenever there is an update, would I have to go through this process each time, or will my choice stick

The modifications will stick, because you'll apply them in your user directory. So unless you change the directory (e.g. new computer), there will be no need to go through the process again.

**Reply by wordmuse** [reply]
Hmm - sorry if I'm a little slow here. I don't see a _user-dir_ under the Freeplane directory. 

I get that "user" is a placeholder. But I'm fully understanding your guidance.

Addendum: I figured it out. So I'm good. 

Thanks a lot for the assist!

~ Bal
